src/Darryldecode/Cart/Cart.php

```diff
@@ -641,11 +640,15 @@ protected function validate($item)
      */
     protected function addRow($id, $item)
     {
+        $this->events->fire($this->getInstanceName().'.adding', array($item, $this));
+
         $cart = $this->getContent();
 
         $cart->put($id, new ItemCollection($item));
 
         $this->save($cart);
+
+        $this->events->fire($this->getInstanceName().'.added', array($item, $this));
     }
 
     /**
```

Comment:
@Beaudinn , so this changes is just a matter of preference right? By the way PR's are welcome..